client/native/storedRecording.go

```diff
@@ -54,6 +53,25 @@ func (sr *StoredRecording) Data(key *ari.Key) (*ari.StoredRecordingData, error)
 	return data, nil
 }
 
+// Download retrieves the data for the stored recording
+// IMPORTANT: Don't forget to close the reader when done.
+func (sr *StoredRecording) Download(key *ari.Key) (*ari.StoredRecordingBinaryData, error) {
```

Comment:
This presents a quandary:  it's too easy to forget to close the body's ReadCloser.

I see three ways to mitigate this, each with their own trade-offs:
 - rename `Download` to `ReadCloser`, in an attempt to make it clear that it must be Closed
 - take an `io.Writer` as a second parameter to `Download()`
 - store the download to a temp file
 
Personally, I think the safest and cleanest is to pass in an `io.Writer`; let the caller indicate where the data should go, so it can just be Piped directly there, and we can close the Body within `Download`.